src/app/services/config/config.module.ts

```diff
@@ -1,26 +1,29 @@
-import { APP_INITIALIZER, NgModule } from "@angular/core";
-import {
-  API_CONFIG,
-  API_ROOT,
-  AppInitializer,
-} from "@helpers/app-initializer/app-initializer";
+import { HttpBackend } from "@angular/common/http";
+import { APP_INITIALIZER, NgModule, Optional } from "@angular/core";
+import { AppInitializer } from "@helpers/app-initializer/app-initializer";
 import { ToastrModule } from "ngx-toastr";
 import { IS_SERVER_PLATFORM } from "src/app/app.helper";
 import { ConfigService } from "./config.service";
+import { API_CONFIG, API_ROOT } from "./config.tokens";
 
 @NgModule({
   imports: [ToastrModule],
   providers: [
     {
       provide: APP_INITIALIZER,
       useFactory: AppInitializer.initializerFactory,
+      deps: [
+        [new Optional(), API_CONFIG],
```

Comment:
what does this mean? I am unfamiliar with this syntax

Reply:
This is an optional DI input. In this case, only SSR declares the API_CONFIG injection token as it does not need to make an api request for the config.